contract_manager/scripts/upgrade_ton_contract.ts

```diff
@@ -0,0 +1,88 @@
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { DefaultStore, loadHotWallet } from "../src";
+import { TonChain } from "../src/chains";
+import { CHAINS, toChainName } from "@pythnetwork/xc-admin-common";
+import fs from "fs";
+import path from "path";
+
+const parser = yargs(hideBin(process.argv))
+  .usage(
+    "Upgrades the Pyth contract on TON and creates a governance proposal for it.\n" +
+      "Usage: $0 --network <mainnet|testnet> --contract-address <address> --ops-key-path <ops_key_path>"
+  )
+  .options({
+    network: {
+      type: "string",
+      choices: ["mainnet", "testnet"],
+      description: "Network to deploy to",
+      demandOption: true,
+    },
+    "contract-address": {
+      type: "string",
+      description: "Address of the contract to upgrade",
+      demandOption: true,
+    },
+    "ops-key-path": {
+      type: "string",
+      description: "Path to operations key file",
+      demandOption: true,
+    },
+  });
+
+async function main() {
+  const argv = await parser.argv;
+  const isMainnet = argv.network === "mainnet";
+
+  // Get chain ID and name from CHAINS mapping
+  const chainId = isMainnet ? CHAINS.ton_mainnet : CHAINS.ton_testnet;
+  const wormholeChainName = toChainName(chainId);
+
+  // Get the TON chain instance with appropriate RPC URL based on network
+  const chain = new TonChain(
+    chainId.toString(),
+    isMainnet,
+    wormholeChainName,
+    undefined,
+    isMainnet
+      ? "https://toncenter.com/api/v2/jsonRPC"
+      : "https://testnet.toncenter.com/api/v2/jsonRPC"
+  );
+
+  const vault =
+    DefaultStore.vaults[
+      "mainnet-beta_FVQyHcooAtThJ83XFrNnv74BcinbRH3bRmfFamAHBfuj"
+    ];
+
+  console.log(
+    `Upgrading contract on TON ${argv.network} (Chain ID: ${chainId}, Wormhole Chain Name: ${wormholeChainName})`
+  );
+
+  // Read the compiled contract from the build directory
+  // NOTE: Remember to build the contract first before running this script to get the latest build, command: `npm run build:blueprint` on target_chains/ton/contracts
+  const compiledPath = path.resolve(
```

Comment:
hmmmm if blueprint:build problem is resolved (the fact that it was interactive in the past) we can add to `build:blueprint` or something and make sure it's a dependency of contract manager so in be in the way of turbo we can get it built by running `pnpm turbo build --filter '@pythnetwork/contract-manager'`